mpc_path_planning_node:
  ros__parameters:
    mpc_path_planning:
      topic_name:
        map: "/map"
        target: "/goal_pose"
        odom: "/odom"
        imu: "/imu"
        global_path: "global_path_planning/path"
        init_path: "mpc_path_planning/init_path"
        opti_path: "mpc_path_planning/opti_path"
        opti_twists: "mpc_path_planning/twists"
      tf_frame:
        map_frame: "map"
        robot_frame: "base_link"
      planning_period: 0.001 #[s]
      mpc:
        dt: 0.1 #[s]
        horizon_time: 5.0
        pos_error: 1000000.0 #[m] 0.8
        # ロボット制約
        forward_only: false
        min_vel: 0.0 #[m/s]  最小速度制限(0~)
        max_vel: 0.55 #[m/s]
        min_angular: 0.0 #[rad/s] 最小速度制限(0~)
        max_angular: 2.84 #[rad/s] 2.84
        max_acc: 2.9 #[m/s^s] 1.1 4.9 3.9 0.98 2.9
        max_angular_acc: 4.9 #[rad/s^s] 1.0 4.9
        # モデルパラメータ
        xy_vel_time_constant: 0.01
        theta_vel_time_constant: 0.01
        #重み[vx,vy,w,x,y,x]
        weight:
          state: [0.5, 0.0, 2.0, 20.0, 20.0, 0.5] #ステートコストの重み [10.0, 0.0, 10.0, 800.0, 800.0, 10.0]
          final_state: [0.5, 0.0, 0.5, 60.0, 60.0, 30.0] #終端コストの重み     [10.0, 0.0, 10.0, 800.0, 800.0, 10.0]
          ref_state: [0.0, 0.0, 0.0, 0.0, 0.0, 0.0] #初期経路(grid_path)とのエラーコストの重み
          control: [1.0, 0.0, 1.0] #制御量コストの重み
          diff_control: [12.0, 0.0, 10.0] #制御量変化コストの重み
        warm_start:
          latest_gpl_norm: 0.1 #[m] 最後に初期値計算を行った地点からlatest_gpl_norm[m]離れたら
          latest_target_norm: 0.01 #[m] 目標値がlatest_target_norm[m]以内(変わってない)とき
        terminal_range: 0.6
        ipopt: #ソルバー設定
          sb: "yes" # コンソールにヘッダを出力しない
          linear_solver: "ma97" # mumpsは遅い ma27,ma57,ma86,ma97はHSLライブラリ必要 (mumps:40msくらい,ma27:10msくらい)
          max_iter: 1000 #最大反復回数 def 500
          acceptable_tol: 0.000001 #収束許容値 def 0.000001
          compl_inf_tol: 0.0001 #相補性条件の「許容」しきい値 def 0.0001
      obstacle_detect:
        dist: 4.0 #[m]
        obstacle_size: 0.5 #[m]
        list_size: 25
        nearby_obstacle_limit: 0.8 #[m]

global_path_planning_node:
  ros__parameters:
    global_path_planning:
      topic_name:
        map: "/map"
        target: "/goal_pose"
        global_path: "global_path_planning/path"
      tf_frame:
        map_frame: "map"
        robot_frame: "base_link"
      planning_period: 0.1 #[s]
      path_planner: "extension_a_star" #dijkstra,dijkstra_dist_map,a_star,extension_a_star,wave_propagation,wave_propagation_dist_map

control_node:
  ros__parameters:
    control:
      topic_name:
        opti_path: "mpc_path_planning/opti_path"
        opti_twists: "mpc_path_planning/twists"
        target: "/goal_pose"
        cmd_vel: "cmd_vel"
      tf_frame:
        map_frame: "map"
        robot_frame: "base_link"
      control_period: 0.001 #[s]
      mpc_dt: 0.1
      goal:
        pos_range: 0.05 #[m] 0.05
        angle_range: 1.0 #[deg] 0.5
        min_vel_range: 0.5 #[m/s] 0.005
        min_angular_range: 0.5 #[rad/s] 0.01
